flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java

```diff
@@ -220,6 +229,9 @@ public void open() {
                 line = lineReader.readLine(prompt, null, inputTransformer, null);
             } catch (UserInterruptException e) {
                 // user cancelled line with Ctrl+C
+                if (!isInteractive) {
```

Comment:
Add a test case about this. The logic we want to verify is:
1. If user types CTRL-C in interactive mode, it only cancelled one line
2. If user types CTRL-C in non-interactive mode, it cancelled the whole CLI.

Reply:
Add test `CliClientTest#testCancelExecutionInNonInteractiveMode` and `CliClientTest#testCancelExecutionInInteractiveMode`